### PR TITLE
Bump version

### PR DESCRIPTION
**Pull request by Wovchena**

### CODE REVIEW

File: .github/workflows/causal_lm_cpp.yml
Comment (Wovchena) [inline .github/workflows/causal_lm_cpp.yml:1]: Reported in ticket 144823.

File: .github/workflows/lcm_dreamshaper_cpp.yml [hunk 1]
```diff
@@ -42,14 +42,15 @@ jobs:
         run: |
           conda activate openvino_lcm_cpp
           conda update -c conda-forge --all
-          conda install -c conda-forge -c conda-forge/label/openvino_dev openvino==2024.2.0.dev20240513 c-compiler cxx-compiler git make cmake
+          conda install -c conda-forge -c conda-forge/label/openvino_dev openvino==2024.2.0 c-compiler cxx-compiler git make cmake
```
Comment (ilya-lavrenov) [inline .github/workflows/lcm_dreamshaper_cpp.yml:45]: as we use official release, we need to drop conda-forge/label/openvino_dev

File: .github/workflows/causal_lm_cpp.yml [hunk 2]
```diff
@@ -486,13 +486,13 @@ jobs:
       - name: Install OpenVINO
         run: |
           mkdir ./ov/
-          curl https://storage.openvinotoolkit.org/repositories/openvino/packages/pre-release/2024.2.0rc2/linux/l_openvino_toolkit_ubuntu20_2024.2.0.dev20240529_x86_64.tgz | tar --directory ./ov/ --strip-components 1 -xz
+          curl https://storage.openvinotoolkit.org/repositories/openvino/packages/nightly/2024.3.0-15766-dd43b82c058/l_openvino_toolkit_ubuntu20_2024.3.0.dev20240621_x86_64.tgz | tar --directory ./ov/ --strip-components 1 -xz
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:489]: can we define version in a single place at the beginning? via env vars

File: .github/workflows/causal_lm_cpp.yml [hunk 3]
```diff
@@ -197,17 +200,15 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: 3.8
-      - name: Install OpenVINO
+      - run: curl --output ov.zip ${{ env.w_ov_link }}
+      - run: unzip -d ov ov.zip
+      - run: dirs=(ov/*) && mv ov/*/* ov && rmdir "${dirs[@]}"
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:205]: can we combine such ugly commands a single one?
```
run: |
 cmd_1
 cmd_2
 ...
 cmd_n
```

File: .github/workflows/lcm_dreamshaper_cpp.yml [hunk 4]
```diff
@@ -42,7 +42,7 @@ jobs:
         run: |
           conda activate openvino_lcm_cpp
           conda update -c conda-forge --all
-          conda install -c conda-forge -c conda-forge/label/openvino_dev openvino==2024.2.0.dev20240513 c-compiler cxx-compiler git make cmake
+          conda install -c conda-forge -c conda-forge/label/openvino_dev openvino==2024.3.0.dev20240614 c-compiler cxx-compiler git make cmake
```
Comment (ilya-lavrenov) [inline .github/workflows/lcm_dreamshaper_cpp.yml:45]: let's wait for https://github.com/conda-forge/openvino-feedstock/runs/26751639883 to complete

Comment (Wovchena) [reply .github/workflows/lcm_dreamshaper_cpp.yml:45]: Passed

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**